libtenzir/builtins/formats/kv.cpp

```diff
@@ -23,10 +24,73 @@ namespace {
 
 constexpr auto docs = "https://docs.tenzir.com/formats/kv";
 
+/// @brief Checks whether the index `idx` in `text` is escaped
+/// TODO the precise-parsers PR contains this in string.hpp
+inline auto is_escaped(size_t idx, std::string_view text) -> bool {
+  if (idx >= text.size()) {
+    return false;
+  }
+  // An odd number of preceding backslashes means it is escaped, for example:
+  // `x\n` => true, `x\\n` => false, `x\\\n` => true, 'x\\\\n' => false.
+  auto backslashes = size_t{0};
+  while (idx > 0 and text[idx - 1] == '\\') {
+    ++backslashes;
+    --idx;
+  }
+  return backslashes % 2 == 1;
+}
+
+[[nodiscard]] auto trim_quotes(std::string_view txt) -> std::string_view {
+  if (txt.size() < 2) {
+    return txt;
+  }
+  if (txt.front() != txt.back()) {
+    return txt;
+  }
+  if (txt.front() == '\"' and not is_escaped(txt.size() - 1, txt)) {
+    txt.remove_prefix(1);
+    txt.remove_suffix(1);
+    return txt;
+  }
+  return txt;
+}
+
+std::string unescape(std::string_view value) {
+  std::string result;
+  result.reserve(value.size());
+  for (auto i = 0u; i < value.size(); ++i) {
+    if (value[i] != '\\') {
+      result += value[i];
+    } else if (i + 1 < value.size()) {
+      auto next = value[i + 1];
+      switch (next) {
+        default: // Anything that isnt a defined escape sequence remains
+          result += '\\';
+          result += next;
+          break;
+        case '\\': // Double backslashes are collapsed
+          result += '\\';
+          break;
+        case '\"': // Escaped quotes are unescaped
+          result += '\"';
+          break;
+      }
+      ++i;
+    }
```

Comment:
This discards backslashes at `value.size() - 1`.

Reply:
We can consider just using JSON escaping/unescaping for this.

Reply:
Agreed (also solves the conversation below this one).